Makefile

```diff
@@ -65,52 +95,58 @@ create_cluster:
 	# for virsh commands in .ssh/authorized_keys
 	set -e;                                               \
 	if ! grep -qf id_rsa.pub ~/.ssh/authorized_keys; then \
```

Comment:
Has anyone confirmed this works locally?

Reply:
I don't think this is a change made in this patch, presumably this was working with libvirt when the change was made in the first place?

Reply:
Was more wondering if anyone has confirmed this works locally

Reply:
I have tested the basics of sshing from the guest to the host with this, yes.  Whether we have any tests that actually will utilise it is another question.

Reply:
Ok. I am curious how this would work when it tries to ssh back to the host as root.

Reply:
It doesn't.  It's not expected to ssh as root.  One does not need to be root to manipulate libvirt VMs if one is in the libvirt group, which the libvirt support here assumes.

Yes, this is not quite compatible with how our tests work, but that's a failing of our tests, not this configuration.  Our tests should be configurable as who they will ssh as.

Reply:
Perhaps I'm missing it, but the setup looks like it uses root as username and lustre as password?

https://github.com/intel-hpdd/intel-manager-for-lustre/blob/0a01be4b7b0b832c160ecebf1859bf9970dc77a9/chroma-manager/tests/framework/integration/shared_storage_configuration/full_cluster/shared_storage_configuration_cluster_cfg.json#L102

Then it looks like the templates use that username + the key to ssh back to the host?

Reply:
[chroma-manager/tests/framework/integration/shared_storage_configuration/full_cluster/shared_storage_configuration_cluster_cfg.json](https://github.com/intel-hpdd/intel-manager-for-lustre/blob/0a01be4b7b0b832c160ecebf1859bf9970dc77a9/chroma-manager/tests/framework/integration/shared_storage_configuration/full_cluster/shared_storage_configuration_cluster_cfg.json)  is not used for SSI on Vagrant.

Instead SSI on Vagrant uses [provisioner_output-vagrant-ssi.json](https://github.com/intel-hpdd/intel-manager-for-lustre/blob/0a01be4b7b0b832c160ecebf1859bf9970dc77a9/provisioner_output-vagrant-ssi.json).

Ideally, that file should be converted/derived from [chroma-manager/tests/framework/integration/shared_storage_configuration/full_cluster/shared_storage_configuration_cluster_cfg.json](https://github.com/intel-hpdd/intel-manager-for-lustre/blob/0a01be4b7b0b832c160ecebf1859bf9970dc77a9/chroma-manager/tests/framework/integration/shared_storage_configuration/full_cluster/shared_storage_configuration_cluster_cfg.json) but that was more work than I had time for at the time.

But you will notice that in [provisioner_output-vagrant-ssi.json](https://github.com/intel-hpdd/intel-manager-for-lustre/blob/0a01be4b7b0b832c160ecebf1859bf9970dc77a9/provisioner_output-vagrant-ssi.json#L10) that I changed the `default_username` so that the ssh key mechanism will work.  That change gives me confidence that the ssh key mechanism actually works.

But yes, this whole process needs some attention.